src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java

```diff
@@ -283,49 +289,49 @@ public static CoordinateArrayListSequence makeCoordinates(Edge[] edges) {
             throw new TrivialPathException();
         }
 
-        int[] legIndexPairs = {0, states.length - 1};
-        List<int[]> legsIndexes = new ArrayList<int[]>();
+        // interval legIndexPairs[0], legIndexPairs[1] contains valid states such as WALK, CAR, TRAIN
+        // interval legIndexPairs[1], legIndexPairs[2] contains LEG_SWITCH states which separate leg from next leg
+        int[] legIndexPairs = {0, states.size()-1, states.size()-1};
```

Comment:
dwa razy ta sama wartość?

Reply:
ta "strukturka" opisuje interwały. od 0... to sate.size() [exclusive]  i od state.size() do state.size() [exclusive]. Rzecz w tym, że te wartości są nadpisywane.
Pierwszy interwał ma zawierać dobre stant z oryginalnej kodu. Drgui interwał zawira stany które normalnie są wycinane, ale ja przechowuje je na wypadek gdyby miały one  jakąś geometrię wartą rysowania

Reply:
brzmi to dość dziwnie, ale okej